src/components/MoneyReportHeader.tsx

```diff
@@ -1,91 +1,63 @@
 import SettlementButton from './SettlementButton';
-import withWindowDimensions, {windowDimensionsPropTypes} from './withWindowDimensions';
-
-const propTypes = {
-    /** The report currently being looked at */
-    report: iouReportPropTypes.isRequired,
-
-    /** The policy tied to the money request report */
-    policy: PropTypes.shape({
-        /** Name of the policy */
-        name: PropTypes.string,
-
-        /** Type of the policy */
-        type: PropTypes.string,
-
-        /** The role of the current user in the policy */
-        role: PropTypes.string,
-    }),
 
+type MoneyReportHeaderOnyxProps = {
     /** The chat report this report is linked to */
-    chatReport: reportPropTypes,
+    chatReport: OnyxEntry<Report>;
 
     /** The next step for the report */
-    nextStep: nextStepPropTypes,
+    nextStep: OnyxEntry<ReportNextStep>;
 
-    /** Personal details so we can get the ones for the report participants */
-    personalDetails: PropTypes.objectOf(participantPropTypes).isRequired,
+    /** UserSession info for the currently logged in user. */
+    session: OnyxEntry<Session>;
+}
 
-    /** Session info for the currently logged in user. */
-    session: PropTypes.shape({
-        /** Currently logged in user email */
-        email: PropTypes.string,
-    }),
+type MoneyReportHeaderProps = MoneyReportHeaderOnyxProps & {
+    /** The report currently being looked at */
+    report: Report;
 
-    ...windowDimensionsPropTypes,
-};
+    /** The policy tied to the money request report */
+    policy: Policy;
 
-const defaultProps = {
-    chatReport: {},
-    nextStep: {},
-    session: {
-        email: null,
-    },
-    policy: {},
+    /** Personal details so we can get the ones for the report participants */
+    personalDetails: PersonalDetails
 };
 
-function MoneyReportHeader({session, personalDetails, policy, chatReport, nextStep, report: moneyRequestReport, isSmallScreenWidth}) {
-    const {windowWidth} = useWindowDimensions();
+function MoneyReportHeader({session, personalDetails, policy, chatReport, nextStep, report: moneyRequestReport}: MoneyReportHeaderProps) {
+    const {windowWidth, isSmallScreenWidth} = useWindowDimensions();
     const styles = useThemeStyles();
     const {translate} = useLocalize();
     const reimbursableTotal = ReportUtils.getMoneyRequestReimbursableTotal(moneyRequestReport);
     const isApproved = ReportUtils.isReportApproved(moneyRequestReport);
     const isSettled = ReportUtils.isSettled(moneyRequestReport.reportID);
-    const policyType = lodashGet(policy, 'type');
-    const isPolicyAdmin = policyType !== CONST.POLICY.TYPE.PERSONAL && lodashGet(policy, 'role') === CONST.POLICY.ROLE.ADMIN;
+    const policyType = policy?.type;
+    const isPolicyAdmin = policyType !== CONST.POLICY.TYPE.PERSONAL && policy?.role === CONST.POLICY.ROLE.ADMIN;
     const isGroupPolicy = _.contains([CONST.POLICY.TYPE.CORPORATE, CONST.POLICY.TYPE.TEAM], policyType);
```

Comment:
Hi @mananjadhav, when using js `includes` to remove underscore, I got this error, I am not sure what we should do.
![issue_exp_ts_mig_1](https://github.com/Expensify/App/assets/48285331/8c4c9e8f-8e35-4918-b876-4cf44babcfec)

Reply:
@KrAbhas You can try to use `some`, it should work
```suggestion
    const isGroupPolicy = [CONST.POLICY.TYPE.CORPORATE, CONST.POLICY.TYPE.TEAM].some(type => type === policyType);
```

Reply:
Thanks Viktoryia, I have made the changes. Please review.